fe2-android/app/src/androidTest/java/com/github/dedis/popstellar/ui/home/CameraPermissionFragmentTest.java

```diff
@@ -0,0 +1,102 @@
+package com.github.dedis.popstellar.ui.home;
+
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static com.github.dedis.popstellar.pages.qrcode.CameraPermissionPageObject.allowCameraButton;
+import static com.github.dedis.popstellar.pages.qrcode.CameraPermissionPageObject.getRequestKey;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import android.os.Bundle;
+
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
+
+import com.github.dedis.popstellar.testutils.FragmentScenarioRule;
+import com.github.dedis.popstellar.testutils.MockResultRegistry;
+import com.github.dedis.popstellar.ui.qrcode.CameraPermissionFragment;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/** Test for the CameraPermissionFragment */
+@RunWith(AndroidJUnit4ClassRunner.class)
+public class CameraPermissionFragmentTest {
+
+  private final MockResultRegistry mockRegistry = new MockResultRegistry();
+
+  @Rule
+  public final FragmentScenarioRule<CameraPermissionFragment> fragmentRule =
+      FragmentScenarioRule.launchInContainer(
+          CameraPermissionFragment.class, () -> CameraPermissionFragment.newInstance(mockRegistry));
+
+  @Test
+  public void allowButtonMakesPermissionRequestAndProducesResult() {
+    setupAllowPermission();
+    Receiver<Bundle> receiver = setupResultListener();
+
+    ViewInteraction allowButton = allowCameraButton().check(matches(ViewMatchers.isClickable()));
+    allowButton.perform(click());
+
+    assertTrue("No response were received", receiver.received());
+    assertEquals("The response is not valid", Bundle.EMPTY, receiver.get());
+  }
+
+  @Test
+  public void denyPermissionDoesNothing() {
+    setupDenyPermission();
+    Receiver<Bundle> receiver = setupResultListener();
+
+    ViewInteraction allowButton = allowCameraButton().check(matches(ViewMatchers.isClickable()));
+    allowButton.perform(click());
+
+    assertFalse("A response was received when none was expected", receiver.received());
```

Comment:
The test is supposed to clear all permission before starting.

That's why we need to use a special rule to grant the permissions we need.

Reply:
Is it supposed to clear permission when we run the tests in local or only on sonarcloud ?

Reply:
Both

Reply:
When I run the tests with `gradle connectCheck` it doesn't work

Reply:
It needs to be done with the mock flavor of the app. Otherwise, the orchestrator is not enabled to produce coverage.
Try running `connectedMockDebugAndroidTest`